tests/integration/release/test_release.py

```diff
@@ -49,6 +48,25 @@ def test_row_noising_omit_row_or_do_not_respond(
     run_omit_row_or_do_not_respond_tests(dataset_name, config, original_data, noised_data)
 
 
+def test_column_dtypes(
+    unnoised_dataset: Dataset,
+    noised_data: pd.DataFrame,
+    dataset_name: str,
+    config: dict[str, Any],
+) -> None:
+    """Tests that column dtypes are as expected"""
+    for col_name in noised_data.columns:
+        col = COLUMNS.get_column(col_name)
+        expected_dtype = col.dtype_name
+        if expected_dtype == np.dtype(object):
+            # str dtype is 'object'
+            # Check that they are actually strings and not some other
+            # type of object.
+            actual_types = noised_data[col.name].dropna().apply(type)
```

Comment:
Is using apply here vectorized? I don't think it is, but maybe we don't have any other options?

Reply:
I believe it's not vectorized, but it might be the fastest way according to this answer:

https://stackoverflow.com/questions/55754713/fastest-way-to-find-all-data-types-in-a-pandas-series